src/nvim/api/ui_events.in.h

```diff
@@ -58,6 +58,8 @@ void set_title(String title)
     FUNC_API_SINCE(3);
 void set_icon(String icon)
     FUNC_API_SINCE(3);
+void highlights_changed(Array highlights_changed)
```

Comment:
Indeed "set" is better. The UI event stream tells the current state, not whether something changed or not